### PR TITLE
Hack the cmor tool for KeyCLIM experiments

### PR DESCRIPTION
**Pull request by YanchunHe**
* Add experiments descriptions in the Control Vocabulary
    * use KeyCLIM to replace CMIP6 activity and rename relavent files
    * remove tables directory under two previouslly created experiment folders: hist-piAerOxid and ssp585-piAerOxid
    * update the cmorSetup for KeyCLIM

### CODE REVIEW

**Comment by YanchunHe** [comment]
Hi Lise and Jerry!

I added all the KeyCLIM experiments in the `tables/CMIP6_CV.json` file, and made other corresponding changes.

Could you reviewer the experiment descriptions under the `experiment_id` section in the `CMIP6_CV.json` in the following file? (you don't need to review the other files).

https://github.com/NorESMhub/noresm2cmor/pull/308/files#diff-0136e9a374b638a4600b6d99ffe8e0c12b9cf242814d97971042aaffb8baafe7

Please note the `piControl-HR`, `hist-HR` and `ssp585-HR` are not included in the `CMIP6_CV.json`, as these experiments can just use `piControl, historical, ssp585` in combination with the `NorESM2-HR` model version.

Further more, here is an example of the global attributes in the cmorized netcdf file of experiment `hist-all`:
```
// global attributes:
		:_NCProperties = "version=1|netcdflibversion=4.6.1|hdf5libversion=1.10.2" ;
		:Conventions = "CF-1.7 CMIP-6.2" ;
		:activity_id = "KeyCLIM" ;
		:branch_method = "Hybrid-restart from year 1200-01-01 of piControl" ;
		:branch_time = 0. ;
		:branch_time_in_child = 0. ;
		:branch_time_in_parent = 438000. ;
		:contact = "Please send any requests or bug reports to noresm-ncc@met.no." ;
		:creation_date = "2021-09-07T13:05:52Z" ;
		:data_specs_version = "01.00.32" ;
		:experiment = "historical simulation with all improvements from other SP experiments" ;
		:experiment_id = "hist-all" ;
		:forcing_index = 1 ;
		:frequency = "fx" ;
		:further_info_url = "https://noresm-docs.readthedocs.io/en/latest/data/keyclim_data.html" ;
		:grid = "finite-volume grid with 0.9x1.25 degree lat/lon resolution" ;
		:grid_label = "gn" ;
		:history = "2021-09-07T13:05:52Z ; CMOR rewrote data to be consistent with CMIP6, CF-1.7 CMIP-6.2 and CF standards." ;
		:initialization_index = 1 ;
		:institution = "NorESM Climate modeling Consortium consisting of CICERO (Center for International Climate and Environmental Research, Oslo 0349), MET-Norway (Norwegian Meteorological Institute, Oslo 0313), NERSC (Nansen Environmental and Remote Sensing Center, Bergen 5006), NILU (Norwegian Institute for Air Research, Kjeller 2027), UiB (University of Bergen, Bergen 5007), UiO (University of Oslo, Oslo 0313) and UNI (Uni Research, Bergen 5008), Norway. Mailing address: NCC, c/o MET-Norway, Henrik Mohns plass 1, Oslo 0313, Norway" ;
		:institution_id = "NCC" ;
		:mip_era = "CMIP6" ;
		:model_id = "NorESM2-MM" ;
		:nominal_resolution = "100 km" ;
		:parent_activity_id = "KeyCLIM" ;
		:parent_experiment_id = "piControl-all" ;
		:parent_mip_era = "CMIP6" ;
		:parent_source_id = "NorESM2-MM" ;
		:parent_sub_experiment_id = "none" ;
		:parent_time_units = "days since 0001-01-01" ;
		:parent_variant_label = "r1i1p1f1" ;
		:physics_index = 1 ;
		:product = "model-output" ;
		:realization_index = 1 ;
		:realm = "atmos" ;
		:run_variant = "N/A" ;
		:source = "NorESM2-MM (2017): \n",
			"aerosol: OsloAero\n",
			"atmos: CAM-OSLO (1 degree resolution; 288 x 192; 32 levels; top level 3 mb)\n",
			"atmosChem: OsloChemSimp\n",
			"land: CLM\n",
			"landIce: CISM\n",
			"ocean: MICOM (1 degree resolution; 360 x 384; 70 levels; top grid cell minimum 0-2.5 m [native model uses hybrid density and generic upper-layer coordinate interpolated to z-level for contributed data])\n",
			"ocnBgchem: HAMOCC\n",
			"seaIce: CICE" ;
		:source_id = "NorESM2-MM" ;
		:source_type = "AOGCM AER BGC" ;
		:sub_experiment = "none" ;
		:sub_experiment_id = "none" ;
		:table_id = "fx" ;
		:table_info = "Creation Date:(28 May 2020) MD5:cb8c5f1b4a2fce62b6cb3cab32c72e91" ;
		:title = "NorESM2-MM output prepared for CMIP6" ;
		:tracking_id = "hdl:21.14100/dbc2ccc4-ce0d-49ed-ba23-cb3afdd2b62b" ;
		:variable_id = "areacella" ;
		:variant_label = "r1i1p1f1" ;
		:license = "KeyCLIM model data produced by NCC is licensed under a Creative Commons Attribution ShareAlike 4.0 International License (https://creativecommons.org/licenses). Consult noresm-ncc@met.no for terms of use, including citation requirements and proper acknowledgment. Further information about this data, including some limitations, can be found via the further_info_url (recorded as a global attribute in this file). The data producers and data providers make no warranty, either express or implied, including, but not limited to, warranties of merchantability and fitness for a particular purpose. All liabilities arising from the supply of the information (including any liability arising in negligence) are excluded to the fullest extent permitted by law." ;
		:cmor_version = "3.6.0" ;
}

```

**Comment by lisesg** [comment]
Thanks @YanchunHe! This looks good! :)

A couple of details: 
- In the CMIP6_CV.json file, we're using using underscores instead of hyphens for the piSpinUp experiments (for instance piSpinUp_ozone), should we consider changing that for consistency (so hyphens in all cases)? I can update the experiment spreadsheet to match this.
- In the global attributes: Can the title be "NorESM2-MM output prepared for KeyCLIM"?

File: tables/CMIP6_CV.json [hunk 1]
```diff
@@ -468,22 +545,22 @@
...
+                    "KeyCLIM"
                 ],
                 "additional_allowed_model_components":[
                     "AER",
                     "CHEM",
                     "BGC"
                 ],
-                "experiment":"all-forcing simulation of the recent past",
-                "experiment_id":"NonCMIP-test",
+                "experiment":"piControl simulation but without with improved ozone + VSLS",
```
Comment (tjiputra) [inline tables/CMIP6_CV.json:557]: remove "without"

File: tables/CMIP6_CV.json [hunk 2]
```diff
@@ -492,17 +569,593 @@
...
+            "hist-pblAtm":{
+                "activity_id":[
+                    "KeyCLIM"
+                ],
+                "additional_allowed_model_components":[
+                    "AER",
+                    "CHEM",
+                    "BGC"
+                ],
+                "experiment":"historical simulation but with atm. BPL perturbation",
```
Comment (tjiputra) [inline tables/CMIP6_CV.json:653]: PBL

File: tables/CMIP6_CV.json [hunk 3]
```diff
@@ -492,17 +569,593 @@
...
+            "ssp585-pblAtm":{
+                "activity_id":[
+                    "KeyCLIM"
+                ],
+                "additional_allowed_model_components":[
+                    "AER",
+                    "CHEM",
+                    "BGC"
+                ],
+                "experiment":"ssp585 simulation but with atm. BPL perturbation",
```
Comment (tjiputra) [inline tables/CMIP6_CV.json:677]: PBL

File: tables/CMIP6_CV.json [hunk 4]
```diff
@@ -492,17 +569,593 @@
...
+            "hist-pblOcn":{
+                "activity_id":[
+                    "KeyCLIM"
+                ],
+                "additional_allowed_model_components":[
+                    "AER",
+                    "CHEM",
+                    "BGC"
+                ],
+                "experiment":"historical simulation but with ocn. BPL perturbation",
```
Comment (tjiputra) [inline tables/CMIP6_CV.json:725]: PBL

File: tables/CMIP6_CV.json [hunk 5]
```diff
@@ -492,17 +569,593 @@
...
+            "ssp585-pblOcn":{
+                "activity_id":[
+                    "KeyCLIM"
+                ],
+                "additional_allowed_model_components":[
+                    "AER",
+                    "CHEM",
+                    "BGC"
+                ],
+                "experiment":"ssp585 simulation but with ocn. BPL perturbation",
```
Comment (tjiputra) [inline tables/CMIP6_CV.json:749]: PBL

File: tables/CMIP6_CV.json [hunk 6]
```diff
@@ -492,17 +569,593 @@
...
+            "piControl-all":{
+                "activity_id":[
+                    "KeyCLIM"
+                ],
+                "additional_allowed_model_components":[
+                    "AER",
+                    "CHEM",
+                    "BGC"
+                ],
+                "experiment":"piControl simulation with all improvements from other SP experiments",
```
Comment (tjiputra) [inline tables/CMIP6_CV.json:1061]: ... with "modifications" from other SP ....

File: tables/CMIP6_CV.json [hunk 7]
```diff
@@ -492,17 +569,593 @@
...
+            "hist-all":{
+                "activity_id":[
+                    "KeyCLIM"
+                ],
+                "additional_allowed_model_components":[
+                    "AER",
+                    "CHEM",
+                    "BGC"
+                ],
+                "experiment":"historical simulation with all improvements from other SP experiments",
```
Comment (tjiputra) [inline tables/CMIP6_CV.json:1085]: ... with "modifications" from other SP ....

File: tables/CMIP6_CV.json [hunk 8]
```diff
@@ -492,17 +569,593 @@
...
+            "ssp126-all":{
+                "activity_id":[
+                    "KeyCLIM"
+                ],
+                "additional_allowed_model_components":[
+                    "AER",
+                    "CHEM",
+                    "BGC"
+                ],
+                "experiment":"ssp126 simulation with all improvements from other SP experiments",
```
Comment (tjiputra) [inline tables/CMIP6_CV.json:1109]: ... with "modifications" from other SP ....

File: tables/CMIP6_CV.json [hunk 9]
```diff
@@ -492,17 +569,593 @@
...
+            "sspt585-all":{
+                "activity_id":[
+                    "KeyCLIM"
+                ],
+                "additional_allowed_model_components":[
+                    "AER",
+                    "CHEM",
+                    "BGC"
+                ],
+                "experiment":"ssp585 simulation with all improvements from other SP experiments",
```
Comment (tjiputra) [inline tables/CMIP6_CV.json:1133]: ... with "modifications" from other SP ....

**Comment by tjiputra** [comment]
thank you very much @YanchunHe. In addition to @lisesg 's suggestion, here are a couple more:
institution = replace "UNI (Uni Research" with "NORCE (Norwegian Research Centre" 
source = replace "MICOM" with "BLOM"

I have also made some minor comments to the CMIP6_CV.json above. I hope I did it correctly.
many thanks, Yanchun.

**Comment by YanchunHe** [comment]
Hi Jerry and Lise,

Thanks for the comments! I have made modifications accordingly. 

Please see the revised [CMIP6_CV.json](https://github.com/NorESMhub/noresm2cmor/commit/2fa4795805a42fe5dfd34f67a33a25d6e24f2e60#diff-0136e9a374b638a4600b6d99ffe8e0c12b9cf242814d97971042aaffb8baafe7)

The final global attributes will look like this:

```log
// global attributes:
		:_NCProperties = "version=1|netcdflibversion=4.6.1|hdf5libversion=1.10.2" ;
		:Conventions = "CF-1.7 CMIP-6.2" ;
		:activity_id = "KeyCLIM" ;
		:branch_method = "Hybrid-restart from year 1200-01-01 of piControl" ;
		:branch_time = 0. ;
		:branch_time_in_child = 0. ;
		:branch_time_in_parent = 438000. ;
		:contact = "Please send any requests or bug reports to noresm-ncc@met.no." ;
		:creation_date = "2021-09-09T13:01:47Z" ;
		:data_specs_version = "01.00.32" ;
		:experiment = "historical simulation but without anthropogenic aerosols" ;
		:experiment_id = "hist-piAerOxid" ;
		:forcing_index = 1 ;
		:frequency = "fx" ;
		:further_info_url = "https://noresm-docs.readthedocs.io/en/latest/data/keyclim_data.html" ;
		:grid = "finite-volume grid with 0.9x1.25 degree lat/lon resolution" ;
		:grid_label = "gn" ;
		:history = "2021-09-09T13:01:47Z ; CMOR rewrote data to be consistent with CMIP6, CF-1.7 CMIP-6.2 and CF standards." ;
		:initialization_index = 1 ;
		:institution = "NorESM Climate modeling Consortium consisting of CICERO (Center for International Climate and Environmental Research, Oslo 0349), MET-Norway (Norwegian Meteorological Institute, Oslo 0313), NERSC (Nansen Environmental and Remote Sensing Center, Bergen 5007), NILU (Norwegian Institute for Air Research, Kjeller 2027), UiB (University of Bergen, Bergen 5007), UiO (University of Oslo, Oslo 0313) and NORCE (Norwegian Research Center, Bergen 5007), Norway. Mailing address: NCC, c/o MET-Norway, Henrik Mohns plass 1, Oslo 0313, Norway" ;
		:institution_id = "NCC" ;
		:mip_era = "CMIP6" ;
		:model_id = "NorESM2-MM" ;
		:nominal_resolution = "100 km" ;
		:parent_activity_id = "KeyCLIM" ;
		:parent_experiment_id = "piControl-piAerOxid" ;
		:parent_mip_era = "CMIP6" ;
		:parent_source_id = "NorESM2-MM" ;
		:parent_sub_experiment_id = "none" ;
		:parent_time_units = "days since 0001-01-01" ;
		:parent_variant_label = "r1i1p1f1" ;
		:physics_index = 1 ;
		:product = "model-output" ;
		:realization_index = 1 ;
		:realm = "atmos" ;
		:run_variant = "N/A" ;
		:source = "NorESM2-MM (2021): \n",
			"aerosol: OsloAero\n",
			"atmos: CAM-OSLO (1 degree resolution; 288 x 192; 32 levels; top level 3 mb)\n",
			"atmosChem: OsloChemSimp\n",
			"land: CLM\n",
			"landIce: CISM\n",
			"ocean: BLOM (1 degree resolution; 360 x 384; 70 levels; top grid cell minimum 0-2.5 m [native model uses hybrid density and generic upper-layer coordinate interpolated to z-level for contributed data])\n",
			"ocnBgchem: HAMOCC\n",
			"seaIce: CICE" ;
		:source_id = "NorESM2-MM" ;
		:source_type = "AOGCM AER BGC" ;
		:sub_experiment = "none" ;
		:sub_experiment_id = "none" ;
		:table_id = "fx" ;
		:table_info = "Creation Date:(28 May 2020) MD5:3b6ee48d181c64cb0f7d8122eb157296" ;
		:title = "NorESM2-MM output prepared for KeyCLIM" ;
		:tracking_id = "hdl:21.14100/55aa1d08-265e-4a26-af7f-02d361371e6a" ;
		:variable_id = "areacella" ;
		:variant_label = "r1i1p1f1" ;
		:license = "KeyCLIM model data produced by NCC is licensed under a Creative Commons Attribution ShareAlike 4.0 International License (https://creativecommons.org/licenses). Consult noresm-ncc@met.no for terms of use, including citation requirements and proper acknowledgment. Further information about this data, including some limitations, can be found via the further_info_url (recorded as a global attribute in this file). The data producers and data providers make no warranty, either express or implied, including, but not limited to, warranties of merchantability and fitness for a particular purpose. All liabilities arising from the supply of the information (including any liability arising in negligence) are excluded to the fullest extent permitted by law." ;
		:cmor_version = "3.6.0" ;
```

**Review by lisesg — APPROVED** [review]
Looks good! Thanks @YanchunHe :)

**Pull request merged**